### PR TITLE
tpm2: fix build failure without openssl

### PR DESCRIPTION
**Pull request by glance-**

### CODE REVIEW

File: src/shared/tpm2-util.c [hunk 1]
```diff
@@ -4483,6 +4483,7 @@ static const char* tpm2_userspace_event_type_table[_TPM2_USERSPACE_EVENT_TYPE_MA
 
 DEFINE_STRING_TABLE_LOOKUP(tpm2_userspace_event_type, Tpm2UserspaceEventType);
 
+#if HAVE_OPENSSL
 const char *tpm2_userspace_log_path(void) {
```
Comment (bluca) [inline src/shared/tpm2-util.c]: This function is exported, but the declaration in the header is not ifdef'ed, which will probably generate compiler warnings. It's not even used outside this module, so just make it static and remove it from the header

**Review by bluca — APPROVED** [review]

File: src/shared/tpm2-util.c (hunk 1 above)
Comment (bluca) [reply src/shared/tpm2-util.c]: or maybe just move the ifdef below this block

Comment (glance-) [reply src/shared/tpm2-util.c]: I'm fine with either. If you'd like me to update the patch just pick one, or do it yourself whatever way you'd like.

**Comment by glance-** [comment]
This is just the latest in a couple of breakages in the build this way, so I'd suggest adding a CI job which builds some of these combinations to find this kind of issues at the source.

**Pull request merged**